portal/static/js/admin.js

```diff
@@ -1,652 +1,781 @@
-                                 */
-                                $("div.export button").attr("disabled", true);
-                              },
-                              dataType: 'json'
-                          }).done(function(data) {
-                                if (data && data.status) {
-                                  data.status.forEach(function(status) {
-                                      var c = status.consents;
-                                      var a = "", s = "", prevItem = {};
-                                      if (c) {
-                                      c.forEach(function(item) {
-                                          if (!item.consent.deleted && (!prevItem.consent_signed || (prevItem.assessment_status != item.assessment_status)
-                                              || (String(prevItem.consent_signed).substring(0, 10) != String(item.consent.signed).substring(0, 10)))) {
-                                              if (!(/null/.test(item.consent.agreement_url))) {
-                                                var cl = "";
-                                                var sd = tnthDates.formatDateString(item.consent.signed);
-                                                var status = item.assessment_status;
-                                                if (!item.consent.send_reminders) status = "withdrawn";
-                                                switch(String(status).toLowerCase()) {
-                                                    case "completed":
-                                                      cl = "text-success";
-                                                      break;
-                                                    case "withdrawn":
-                                                      cl = "text-muted";
-                                                      break;
-                                                    case "due":
-                                                      cl = "text-warning";
-                                                      break;
-                                                    case "overdue":
-                                                      cl = "text-danger";
-                                                      break;
+    return self.patientsIdList;
+  };
+  /*
+   * currently not being used - client side retrieval of assessment status
+   */
+  AdminTool.prototype.getData = function(requests, callback) {
+      var self = this;
+      if (self.ajaxAborted) {
+          return false;
+      };
+      if (!requests || requests.length === 0) {
+          return false;
+      };
+      var userString = requests.shift();
+      if (!hasValue(userString)) {
+          return false;
+      };
+
+      var tnthDates = self.getDependency("tnthDates");
+      /*
+       *  load the data sequentially
+       *  Note, NO concurrent ajax calls here,
```

Comment:
Seems like a good opportunity to try [Web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) if we're still having performance problems :)

Reply:
yeah, ivan, web work is a great idea.  will try that if we need to get assessment statuses via frontend in the future.  We have switched since to get them from backend.  This function is not in use at the moment.